src/DataStructures/Tensor/Expressions/Contract.hpp

```diff
@@ -178,37 +193,30 @@ template <template <typename> class TE, typename ReplacedArgList, typename I,
           typename TotalContracted>
 struct fully_contract_helper {
```

Comment:
Same question: can the classes business by simplified using `if constexpr`?

Reply:
Done!

Reply:
Some additional comment: the struct can be remove completely and only have a function